src/Storages/StorageFile.cpp

```diff
@@ -526,12 +496,172 @@ namespace
+                    };
+
+                    auto cache_key = getKeyForSchemaCache(full_path, format, format_settings, context);
+                    auto columns = schema_cache.tryGetColumns(cache_key, get_last_mod_time);
+
+                    if (columns)
+                        return columns;
+
+                    return std::nullopt;
+                };
+
+                if (archive_info.readSingleFile())
+                {
+                    read_buf = archive_reader->readFile(archive_info.path_in_archive, false);
+                    ++current_archive_index;
+                    if (!read_buf)
+                        continue;
+
+                    const auto & full_path = processed_files.emplace_back(fmt::format("{}::{}", archive_reader->getPath(), archive_info.path_in_archive));
+                    columns_from_cache = check_schema_cache(full_path);
+
+                    if (columns_from_cache)
+                        return nullptr;
+                }
+                else
+                {
+                    auto & read_files = read_files_from_archive[current_archive_index];
+                    auto file_enumerator = archive_reader->firstFile();
+                    if (!file_enumerator)
+                    {
+                        if (getContext()->getSettingsRef().engine_file_skip_empty_files)
+                        {
+                            ++current_archive_index;
+                            continue;
+                        }
+
+                        throw Exception(
+                            ErrorCodes::CANNOT_EXTRACT_TABLE_STRUCTURE,
+                            "Cannot extract table structure from {} format file, because the archive {} has no files. "
+                            "You must specify table structure manually",
+                            format,
+                            archive);
+                    }
+
+                    const auto * filename = &file_enumerator->getFileName();
+                    while (read_files.contains(*filename) || !archive_info.filter(*filename))
+                    {
+                        if (!file_enumerator->nextFile())
+                        {
+                            archive_reader = nullptr;
+                            break;
+                        }
+
+                        filename = &file_enumerator->getFileName();
+                    }
+
+                    if (!archive_reader)
+                    {
+                        ++current_archive_index;
+                        continue;
+                    }
+
+                    const auto & full_path = processed_files.emplace_back(fmt::format("{}::{}", archive_reader->getPath(), *filename));
+
+                    read_files.insert(*filename);
+                    read_buf = archive_reader->readFile(std::move(file_enumerator));
+
+                    columns_from_cache = check_schema_cache(full_path);
+
+                    if (columns_from_cache)
```

Comment:
If we return nullptr if `columns_from_cache != nullptr`, then local variable `read_buf`, which is just initialized, is not needed, so no need to initialize it here in line 633?